### PR TITLE
whep protocols

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: protobufs/rpc/whep.proto [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        option (psrpc.options).affinity_func = true;
+    };
+}
+
+service WHEPHandler {
+    rpc ICETrickle(ICETrickleRequest) returns(google.protobuf.Empty) {
+        option (psrpc.options) = {
+            topics: true
+            topic_params: {
+                group: "Resuource"
```
Comment (boks1971) [inline protobufs/rpc/whep.proto:22]: `Resource` spelling

File: sdp/sdp.go [hunk 2]
```diff
@@ -149,3 +149,32 @@ func ExtractStreamID(media *sdp.MediaDescription) (string, bool) {
...
+	if err := parsed.UnmarshalString(in); err != nil {
+		return "", err
+	}
+
+	replaceAttributes(parsed.Attributes)
+	for _, m := range parsed.MediaDescriptions {
+		replaceAttributes(m.Attributes)
+	}
+
+	newRemoteDescription, err := parsed.Marshal()
```
Comment (boks1971) [inline sdp/sdp.go:174]: Rename this to `outDescription` or `replacedDescription`. `Remote` does not fit in a util function.

File: rpc/whep_client.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	WHEPHandlerClient
+}
+
+func NewWHEPClient(params ClientParams) (WHEPClient, error) {
+	if params.Bus == nil {
+		return nil, nil
+	}
+	opts := params.Options()
+
+	internalClient, err := NewWHEPInternalClient(params.Bus, opts...)
```
Comment (boks1971) [inline rpc/whep_client.go:33]: looks like maybe missing a file with these types?

**Review by paulwe — APPROVED** [review]

**Comment by davidzhao** [comment]
question, does there need to be an API call for WHEP at all? could it be just starting the session with `https://myurl/whep/room/identity` ?

**Comment by cnderrauber** [comment]
> question, does there need to be an API call for WHEP at all? could it be just starting the session with `https://myurl/whep/room/identity` ?

No, just use the WHEP player to access the url with token

File: utils/guid/id.go [hunk 4]
```diff
@@ -46,7 +46,8 @@ const (
 	SIPDispatchRulePrefix = "SDR_"
 	SIPCallPrefix         = "SCL_"
 	RPCPrefix             = "RPC_"
-	WHIPResourcePrefix    = "WH_"
+	WHIPResourcePrefix    = "WHIP_"
```
Comment (cnderrauber) [inline utils/guid/id.go:49]: separate WHIP and WHEP prefixes. cc @biglittlebigben

**Review by biglittlebigben — APPROVED** [review]

**Comment by cnderrauber** [comment]
TBD

**Pull request closed without merging**